service/resource/s3objectv2/spec.go

```diff
@@ -47,6 +42,7 @@ type S3Client interface {
 	GetObject(*s3.GetObjectInput) (*s3.GetObjectOutput, error)
 	PutObject(*s3.PutObjectInput) (*s3.PutObjectOutput, error)
 	DeleteObject(*s3.DeleteObjectInput) (*s3.DeleteObjectOutput, error)
+	ListObjectsV2(*s3.ListObjectsV2Input) (*s3.ListObjectsV2Output, error)
```

Comment:
Why is there some kind of versioning on the method? I think we should rather version on the very root of a package. Otherwise we end up making version based exceptions inside the code and this would undermine the idea of version bundles again.

Reply:
The V2 here is from AWS. They recommend using ListObjectsV2 for new development so I went with that.

https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#S3.ListObjectsV2